src/PricingModules/UniswapV3/UniswapV3WithFeesPricingModule.sol

```diff
@@ -0,0 +1,442 @@
+            uint24,
+            int24,
+            int24,
+            uint128,
+            uint256,
+            uint256,
+            uint128,
+            uint128
+        ) {
+            return exposure[token0].maxExposure != 0 && exposure[token1].maxExposure != 0;
+        } catch {
+            return false;
+        }
+    }
+
+    /*///////////////////////////////////////////////////////////////
+                          PRICING LOGIC
+    ///////////////////////////////////////////////////////////////*/
+
+    /**
+     * @notice Returns the value of a Uniswap 3 Liquidity Range, denominated in USD.
+     * @param getValueInput A Struct with the input variables (avoid stack too deep).
+     * - asset: The contract address of the asset.
+     * - assetId: The Id of the range.
+     * - assetAmount: The amount of assets.
+     * - baseCurrency: The BaseCurrency in which the value is ideally denominated.
+     * @return valueInUsd The value of the asset denominated in USD, with 18 Decimals precision.
+     * @return valueInBaseCurrency The value of the asset denominated in a BaseCurrency different from USD, with 18 Decimals precision.
+     * @return collateralFactor The collateral factor of the asset for a given baseCurrency, with 2 decimals precision.
+     * @return liquidationFactor The liquidation factor of the asset for a given baseCurrency, with 2 decimals precision.
+     * @dev The UniswapV3PricingModule will always return the value denominated in USD.
+     */
+    function getValue(IPricingModule.GetValueInput memory getValueInput)
+        public
+        view
+        override
+        returns (uint256 valueInUsd, uint256, uint256 collateralFactor, uint256 liquidationFactor)
+    {
+        // Use variables as much as possible in local context, to avoid stack too deep errors.
+        address asset = getValueInput.asset;
+        uint256 id = getValueInput.assetId;
+        uint256 baseCurrency = getValueInput.baseCurrency;
+        address token0;
+        address token1;
+        uint256 usdPriceToken0;
+        uint256 usdPriceToken1;
+        uint256 principal0;
+        uint256 principal1;
+        {
+            int24 tickLower;
+            int24 tickUpper;
+            uint128 liquidity;
+            (,, token0, token1,, tickLower, tickUpper, liquidity,,,,) = INonfungiblePositionManager(asset).positions(id);
+
+            // Uniswap Pools can be manipulated, we can't rely on the current price (or tick).
+            // We use Chainlink oracles of the underlying assets to calculate the flashloan resistant current price.
+            // usdPriceToken is the USD price for 10**tokenDecimals of tokens, it has 18 decimals precision.
+            (usdPriceToken0,,,) = PricingModule(erc20PricingModule).getValue(
+                GetValueInput({ asset: token0, assetId: 0, assetAmount: unit[token0], baseCurrency: 0 })
+            );
+            (usdPriceToken1,,,) = PricingModule(erc20PricingModule).getValue(
```

Comment:
here you already get the risk variables, can be used in line 186/188

Reply:
No sir, these are the risk vars for usd, not baseCurrency